File: Data Structures & Algorithms/max-water-container/submission-2.java
class Solution {
    public int maxArea(int[] heights) {
        int maxWater = 0;

        int left = 0; 
        int right = heights.length-1;

        while(left < right) { 
        int width = (right - left); 
        int height = Math.min(heights[right], heights[left]); 
        int currWater = width * height;

        maxWater = Math.max(maxWater, currWater); 
        if(heights[right] < heights[left]){ 
            right--; 
        } else if(heights[right] > heights[left]){ 
            left++;
        } else { 
            right--; 
        }
    }

    return maxWater;




    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left = 0
        right = left + 1
        res = 0

        while(left <= right and right < len(prices)): 
            res = max(res, prices[left] - prices[right])

            while(prices[left] > prices[right]): 
                left += 1

            right += 1

        return res 